lib/request.js

```diff
@@ -81,10 +83,12 @@ function doRequest (options, data) {
         }
       })
       process.stdin.on('end', async function () {
-        // There will be a trailing \n from the user hitting enter. Get rid of it.
-        data = data.replace(/\n$/, '')
-        const result = await doRequest(options, data)
-        console.log(JSON.stringify(result))
+        try {
+          const result = await doRequest(options, data)
+          console.log(JSON.stringify(result))
+        } catch (e) {
+          console.log(JSON.stringify({ error: e }))
+        }
```

Comment:
This adds some pretty obvious error handling, the need for which was made obvious by the tests. The following catch is in a different function, so this error got swallowed silently.

Reply:
> The following catch is in a different function, so this error got swallowed silently.

Indeed, nice catch :+1: 
Could you please annotate the corresponding test? I will extract this change in a dedicated pull request.